### PR TITLE
Add list of supported tools

### LINKED ISSUE #11187: Docs: create a page per doctool supported
We ended up wanting to expose examples to our users simpler and easy to maintain than our "Example projects" (https://docs.readthedocs.io/en/stable/examples.html) and better and more user-friendly and production-ready than our `test-builds` project (https://test-builds.readthedocs.io/en/latest/).

I think that something _in-between_ those would work pretty well in our documentation. I imagine a page per doctool with:
* one paragraph with the introduction to the doctool and link to the official website
* a YAML configuration file example
* a link to a live documentation example from a specific `test-builds` branch (e.g. https://test-builds.readthedocs.io/en/docsify/)
* explanation of the arguments passed to the command if there are something special for Read the Docs
* extra resources (themes, tutorials, etc)

In fact, I'd like to see something like this section https://docs.readthedocs.io/en/stable/build-customization.html#build-commands-examples but with a little more explanation and following the template proposed previously.

Summarizing, it seems we have 60% of the work done already but we just need to standardize it 👍🏼 

Reference:

* https://github.com/readthedocs/website/pull/268#issuecomment-1977075341
* https://github.com/readthedocs/website/pull/263#issuecomment-1952399024

### PR DESCRIPTION
**Pull request by plaindocs**
And also addon config for each one.

Fixes #11187

Still to do:

- [x] delete Getting started with * content 
- [x] redirect `user/intro/getting-started-with-mkdocs.rst` -> `user/intro/mkdocs`
- [x] redirect `user/intro/getting-started-with-sphinx.rst` -> `user/intro/sphinx`
- [x] figure out example repos
- [x] revert dependencies

Open questions:

- [ ]  we probably need to address the distinction between people using `mkdocs` in the RTD config, and people using `mkdocs` in build commands on this page. And go back to the working here https://docs.readthedocs.io/en/stable/addons.html#enabling-read-the-docs-addons )are you already using the addons, or are the addons already usable if you've configured them) 

---
:books: Documentation previews :books:

- User's documentation (`docs`): https://docs--11547.org.readthedocs.build/en/11547/intro/doctools.html

### CODE REVIEW

File: docs/user/guides/canonical-urls.rst [hunk 1]
```diff
@@ -26,7 +26,7 @@ and letting Read the Docs define it.
 MkDocs
 ~~~~~~
 
-For :doc:`MkDocs </intro/getting-started-with-mkdocs>` we do not define your canonical domain automatically,
+For :doc:`MkDocs </intro/mkdocs>` we do not define your canonical domain automatically,
```
Comment (humitos) [inline docs/user/guides/canonical-urls.rst:29]: We don't define the canonical domain automatically on any tool now. If there are documentation saying that, we should change it and point to the specific guide (the ones we are writing here).

File: docs/user/intro/doctools.rst [hunk 2]
```diff
@@ -0,0 +1,15 @@
+Supported doc tools
+===================
+
+Read the Docs provides hosting for documentation,
+regardless of what tool you use to build it.
```
Comment (humitos) [inline docs/user/intro/doctools.rst:5]: ... as long as it outputs regular HTML files 😄

File: docs/user/intro/doctools.rst [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+Here are minimal configuration examples for some common tools,
+with more coming soon.
+
+.. toctree::
+   :maxdepth: 2
+   :caption: Supported doctools
+
+   Hosting Sphinx documentation on Read the Docs </intro/sphinx>
+   Hosting Material for Mkdocs documentation on Read the Docs </intro/mkdocs>
```
Comment (humitos) [inline docs/user/intro/doctools.rst:15]:
```suggestion
   Hosting Material for MkDocs documentation on Read the Docs </intro/mkdocs>
```

File: docs/user/intro/mkdocs.rst [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+.. _Getting started with Material for MkDocs: https://squidfunk.github.io/mkdocs-material/getting-started/
+
+Configuring Material for MkDocs and Read the Docs addons
+--------------------------------------------------------
+
+To get the best integration with Read the Docs,
+you need to make the following configuration changes to your Material for MkDocs config:
+
+#. Set the site URL using a :doc:`Read the Docs environment variable </reference/environment-variables>`:
```
Comment (humitos) [inline docs/user/intro/mkdocs.rst:52]: We can probably link to `site_url` config here: https://www.mkdocs.org/user-guide/configuration/#site_url

File: docs/user/intro/mkdocs.rst [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+- If you're new to Material for MkDocs, check out the official `Getting started with Material for MkDocs`_ guide.
+
+.. _Getting started with Material for MkDocs: https://squidfunk.github.io/mkdocs-material/getting-started/
+
+Configuring Material for MkDocs and Read the Docs addons
+--------------------------------------------------------
+
+To get the best integration with Read the Docs,
+you need to make the following configuration changes to your Material for MkDocs config:
```
Comment (humitos) [inline docs/user/intro/mkdocs.rst:50]: Is it possible to mention this is _optional_ somehow? Is it currently enough saying "to get the best integration..." to convey this is optional?

File: docs/user/intro/mkdocs.rst [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+#. Make sure that ``javascript/readthedocs.js`` is included in the MkDocs build:
+
+    .. code-block:: yaml
+        :caption: mkdocs.yml
+
+        extra_javascript:
+            - javascript/readthedocs.js
+
+Example repo and demo
```
Comment (humitos) [inline docs/user/intro/mkdocs.rst:137]: I would use the whole word here, 
```suggestion
Example repository and demo
```

File: docs/user/intro/mkdocs.rst (hunk 4 above)
Comment (humitos) [reply docs/user/intro/mkdocs.rst:52]: Also, I think it would be good to explain _what is this for_. Maybe with a small sentence and linking to the https://docs.readthedocs.io/en/stable/guides/canonical-urls.html page.

File: docs/user/intro/mkdocs.rst [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+you need to make the following configuration changes to your Material for MkDocs config:
+
+#. Set the site URL using a :doc:`Read the Docs environment variable </reference/environment-variables>`:
+
+    .. code-block:: yaml
+        :caption: mkdocs.yml
+
+        site_url: !ENV READTHEDOCS_CANONICAL_URL
+
+#. Configure search to use Read the Docs search instead of the default search:
```
Comment (humitos) [inline docs/user/intro/mkdocs.rst:59]: Here as well, we can probably link to https://docs.readthedocs.io/en/stable/server-side-search/index.html

File: docs/user/intro/mkdocs.rst [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+#. Include ``javascript/readthedocs.js`` in the MkDocs build:
+
+    .. code-block:: yaml
+        :caption: mkdocs.yml
+
+        extra_javascript:
+            - javascript/readthedocs.js
+
+
+Integrating the Read the Docs version menu into your site navigation
```
Comment (humitos) [inline docs/user/intro/mkdocs.rst:81]: Is there a specific reason why this is on its own section separated from the addons integration?

Comment (humitos) [reply docs/user/intro/mkdocs.rst:81]: What about the following structure?

* Configuring Material for MkDocs and Read the Docs addons
  * Canonical URL
  * Search
  * Version selector

Each of section with the numbered steps you described if they have more than 1 step.

File: docs/user/intro/mkdocs.rst [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+   :description lang=en: Hosting Material for MkDocs on Read the Docs.
+
+MkDocs is a fast, simple static site generator that's geared towards building project documentation.
+Material for MkDocs is a powerful documentation framework on top of MkDocs.
+
+.. note::
+
+    This page is explicitly about Material for MkDocs. We're working on a guide for plain MkDocs as well.
+
+.. TODO The code comments for this next coe block are pre-addons right? cos there is no manipulation
```
Comment (humitos) [inline docs/user/intro/mkdocs.rst:14]: Yes, that YAML file is old. I can adapt the YAML file from the repository so we can update it here as well 👍🏼

**Review by humitos — COMMENTED** [review]
I think this is going in the correct direction.

I left a suggestion about the section structure for the addons that I think it works better.

Apart from that, there are just other small suggestions to link to external and internal pages.

Let me know if you have any particular doubt where I can be useful.

File: docs/user/guides/canonical-urls.rst (hunk 1 above)
Comment (plaindocs) [reply docs/user/guides/canonical-urls.rst:29]: I was super confused by this for a moment. Right, I only updated the file for the broken link. Probably I should fix the merge resolution first. So thanks for the pointer on that update.

File: docs/user/intro/mkdocs.rst (hunk 8 above)
Comment (plaindocs) [reply docs/user/intro/mkdocs.rst:81]: Yeah, I was trying to avoid that to highlight the functionality not the addons, but looking at it in more depth I think it's the right take.

**Comment by plaindocs** [comment]
@humitos how general is the advice to pin your Sphinx version? Should we be recommending this for all doctools?

```
   We strongly recommend to :doc:`pin the Sphinx version </guides/reproducible-builds>`
   used for your project to build the docs to avoid potential future incompatibilities.
```

**Comment by humitos** [comment]
> @humitos how general is the advice to pin your Sphinx version? Should we be recommending this for all doctools?

It's good advice. However, I don't think we need to cover this into these introduction pages. It will deviate the attention to a completely different and deep hole.

File: docs/user/intro/sphinx.rst [hunk 10]
```diff
@@ -1,26 +1,107 @@
...
 
 Sphinx is a powerful documentation generator that
 has many features for writing technical documentation.
 
+Minimal configuration required to build an existing Sphinx project on Read the Docs looks like this,
+specifying a python 3.x toolchain on Ubuntu, defining the location of the installation requirements, and using the built-in
+:ref:`mkdocs <config-file/v2:sphinx>` command:
+
+
+.. TODO:: why do we install tree using apt_package here? Because it might not be in the requirements.txt but is still necessary?
```
Comment (humitos) [inline docs/user/intro/sphinx.rst]: Yeah, we don't need `tree` here.

File: docs/user/intro/sphinx.rst [hunk 11]
```diff
@@ -1,26 +1,107 @@
...
 .. code-block:: yaml
    :caption: .readthedocs.yaml
 
-   version: 2
-   sphinx:
-      configuration: docs/conf.py
+      version: 2
+
+      build:
+         os: ubuntu-22.04
```
Comment (humitos) [inline docs/user/intro/sphinx.rst:23]: `ubuntu-24.04`

File: docs/user/intro/doctools.rst
Comment (agjohnson) [inline docs/user/intro/doctools.rst:1]: Regarding structure, the only missing piece is probably a little more content on this main listing page. I wouldn't build out a lot of information here, especially not duplicate information from the subpages. But, perhaps there is room for at least a description of each tool, with some basics about each -- input/output formats, language, etc.

Something that would help users select a tool if they aren't familiar with any of these tools. This is a common scenario, more so on the business hosting side.

Netlify does an overview of each framework on this listing page as well, though they put all framework information on one page too.

https://docs.netlify.com/frameworks/

**Review by agjohnson — COMMENTED** [review]
The subpage structure seems plenty good, we can always build out more content on the listing page later but I did note it for now.

File: docs/user/intro/sphinx.rst (hunk 11 above)
Comment (plaindocs) [reply docs/user/intro/sphinx.rst:23]: For this, and `tree` I'm using https://github.com/readthedocs/test-builds/blob/sphinx-7.0.x/.readthedocs.yaml -> is there something more recent I should be using?

File: docs/user/intro/doctools.rst [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
```
[the remark this answers is not in the document]
Comment (plaindocs) [reply docs/user/intro/doctools.rst:15]: Yeah, I was aiming at some of the SEO that Eric mentioned, but that was a few iterations back, happy to revert to shorter.

[the remark this answers is not in the document]
Comment (plaindocs) [reply docs/user/intro/doctools.rst:15]: And there is th meta desc in-page as well.

File: docs/user/intro/doctools.rst
Comment (plaindocs) [reply docs/user/intro/doctools.rst:1]: image

looks pretty good to me

Comment (plaindocs) [reply docs/user/intro/doctools.rst:1]: Though if we're adding that info on the listing page, we really should list it on the detail page as well. :thinking:

Comment (plaindocs) [reply docs/user/intro/doctools.rst:1]: Could install [datacontribtemplates](https://sphinxcontribdatatemplates.readthedocs.io/en/latest/yaml.html) as well, but it's _probably_ overkill for this :-D

**Comment by plaindocs** [comment]
I can easily revert [c7216e9](https://github.com/readthedocs/readthedocs.org/pull/11547/commits/c7216e936a3a99481e0326da48a344937f73c26c) if we'd prefer a different approach.

**Comment by plaindocs** [comment]
At a guess I need to do the `pip-compile --output-file=requirements/docs.txt requirements/docs.in` when adding dependencies to docs.in?

**Comment by humitos** [comment]
> At a guess I need to do the `pip-compile --output-file=requirements/docs.txt requirements/docs.in` when adding dependencies to docs.in?

Yes 😄

File: docs/user/intro/doctools.rst
Comment (humitos) [reply docs/user/intro/doctools.rst:1]: Is it possible to make the whole card clickable? It's hard to realize that the Sphinx word is the link to the subpage. I thought it was just a link to the official Sphinx documentation.

Comment (plaindocs) [reply docs/user/intro/doctools.rst:1]: Ah, yeah, I had the same problem, and actually removed the external link to sphinx to reduce that confusion. 

tl;dr you can't, although I could look at doing something similar with just the design dependency, which we already use, and does have that capability.

Comment (humitos) [reply docs/user/intro/doctools.rst:1]: It looks _way_ better to me 💯 

Screenshot_2024-09-10_17-12-14

File: docs/user/intro/doctools.rst [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+Read the Docs provides hosting for static html documentation generated by Sphinx,
+MkDocs, Docusaurus and a variety of other tools.
+
+Here are minimal configuration examples for some common tools,
+with more coming soon.
+
+.. grid:: 2
+
+    .. grid-item-card::  Material for MkDocs
+        :link: /intro/mkdocs.html
```
Comment (humitos) [inline docs/user/intro/doctools.rst:13]:
```suggestion
        :link: intro/mkdocs.html
```

These links should be relative

Comment (plaindocs) [reply docs/user/intro/doctools.rst:13]: It would make sense, but your suggestion doesn't work. I also attempted to use an RST ref/doc but neither of those worked for me,hence the hard coded html :(

File: docs/user/intro/doctools.rst
Comment (agjohnson) [reply docs/user/intro/doctools.rst:1]: That looks great. I agree I don't want to add too much information on the overview pages, this is striking a good balance for now.

File: docs/user/intro/doctools.rst (hunk 13 above)
Comment (plaindocs) [reply docs/user/intro/doctools.rst:13]: Never mind :man_facepalming:

**Comment by humitos** [comment]
> we probably need to address the distinction between people using `mkdocs` in the RTD config, and people using `mkdocs` in build commands on this page

What is this distinction you refer to here?

File: docs/user/intro/sphinx.rst [hunk 14]
```diff
@@ -0,0 +1,168 @@
...
+You *may* need to set `flyout_display` to `hidden <https://sphinx-rtd-theme.readthedocs.io/en/latest/configuring.html#confval-flyout_display>`_ in your ``conf.py`` so as not to display two identical menus:
+
+.. code-block:: py
+   :caption: conf.py
+
+    html_theme_options = {
+        "flyout_display": "hidden",
+    }
+
+If you're using a different theme, the flyout menu will display in the default bottom right side of your docs.
```
[the remark this answers is not in the document]
Comment (humitos) [reply docs/user/intro/sphinx.rst]: I went ahead and commented this section for now in [eb71007](https://github.com/readthedocs/readthedocs.org/pull/11547/commits/eb71007be8c6261cdf2d67945c3b44d1e3f3667a); but we need to revisit this.

**Comment by humitos** [comment]
I made some small changes to this PR so we can merge it now and making it part of tomorrow's release. However, there are a few things to work on still. I want to make those changes in an iterative way so we can start linking this content to users and testing how it performs.

File: docs/user/intro/sphinx.rst (hunk 14 above)
[the remark this answers is not in the document]
Comment (plaindocs) [reply docs/user/intro/sphinx.rst]: Hey @humitos , let me see if I've got this right:

- version / language selectors are enabled by default (are they different from the new flyout?)  - are they configurable? 
- The flyout is hidden by default, to show it use flyout_display: "attached"

**Comment by plaindocs** [comment]
> > we probably need to address the distinction between people using `mkdocs` in the RTD config, and people using `mkdocs` in build commands on this page
> 
> What is this distinction you refer to here?

So you could (in theory) build mkdocs docs like https://docs.readthedocs.io/en/stable/config-file/v2.html#build-commands or with https://docs.readthedocs.io/en/stable/config-file/v2.html#mkdocs right? 

But would there be any config difference?

File: docs/user/intro/sphinx.rst (hunk 14 above)
[the remark this answers is not in the document]
Comment (humitos) [reply docs/user/intro/sphinx.rst]: > version / language selectors are enabled by default?

Yes, only on our own theme, tho.

> (are they different from the new flyout?)?

Yes and no. They use the exact same data (same versions, languages, etc). However, the version/language selectors shown in the top left nav bar are managed by the theme itself.

On the other hand, the new flyout is automatically injected by Read the Docs and is shown floating a the bottom.

> are they configurable?

The version/language selectors are not configurable. The flyout is configurable via the Addons' settings in the dashboard.

> The flyout is hidden by default, to show it use flyout_display: "attached"

No. The new flyout (the one that Read the Docs injects) is shown by default (together with the version/language selectors from the theme).

If `flyout_display` is defined as `attached` in the theme options, the "integrated flyout" will be shown in at the bottom of the navbar. However, this does nothing on the new Read the Docs Addons flyout. At this point, both flyout will be shown.

The Read the Docs Addons flyout can be disabled from the Addons' settings in the dashboard.

-----

Does this make sense?

[the remark this answers is not in the document]
Comment (plaindocs) [reply docs/user/intro/sphinx.rst]: Let me try to poke at things and see if it does :-D
